x/programs/rust/wasmlanche-sdk/src/memory.rs

```diff
@@ -88,22 +84,100 @@ pub extern "C" fn alloc(len: usize) -> HostPtr {
     HostPtr(ptr.cast_const())
 }
 
+fn remove(ptr: *const u8) -> usize {
+    ALLOCATIONS
+        .with_borrow_mut(|allocations| allocations.remove(&ptr))
+        .expect("attempted to drop invalid host pointer")
+}
+
 #[cfg(test)]
 mod tests {
+    #![allow(clippy::useless_vec)]
     use super::*;
 
     #[test]
-    fn data_allocation() {
-        let len = 1024;
-        let ptr = alloc(len);
-        let vec: Vec<_> = (u8::MIN..=u8::MAX).cycle().take(len).collect();
-        unsafe { std::ptr::copy(vec.as_ptr(), ptr.0.cast_mut(), vec.len()) }
-        assert_eq!(*ptr, vec);
+    #[should_panic(expected = "attempted to deref invalid host pointer")]
```

Comment:
is it standard to hardcode these string error messages?

Reply:
I tried to put them in a constant and the attribute couldn't read the constant. So I'm not sure if there's a better way to do this. The only reason I checked the message in the first place was because Clippy told me to.